tests/components/time_date/test_sensor.py

```diff
@@ -1,172 +1,159 @@
+    assert device.state == "2017-05-18"
+
+    device = time_date.TimeDateSensor(hass, "time_utc")
+    device._update_internal_state(now)
+    assert device.state == "00:54"
+
+    device = time_date.TimeDateSensor(hass, "date_time")
+    device._update_internal_state(now)
+    assert device.state == "2017-05-18, 00:54"
+
+    device = time_date.TimeDateSensor(hass, "date_time_utc")
+    device._update_internal_state(now)
+    assert device.state == "2017-05-18, 00:54"
+
+    device = time_date.TimeDateSensor(hass, "beat")
+    device._update_internal_state(now)
+    assert device.state == "@079"
+
+    device = time_date.TimeDateSensor(hass, "date_time_iso")
+    device._update_internal_state(now)
+    assert device.state == "2017-05-18T00:54:00"
+
+
+async def test_states_non_default_timezone(hass):
+    """Test states of sensors in a timezone other than UTC."""
+    new_tz = dt_util.get_time_zone("America/New_York")
+    assert new_tz is not None
+    dt_util.set_default_time_zone(new_tz)
+
+    now = dt_util.utc_from_timestamp(1495068856)
+    device = time_date.TimeDateSensor(hass, "time")
+    device._update_internal_state(now)
+    assert device.state == "20:54"
+
+    device = time_date.TimeDateSensor(hass, "date")
+    device._update_internal_state(now)
+    assert device.state == "2017-05-17"
+
+    device = time_date.TimeDateSensor(hass, "time_utc")
+    device._update_internal_state(now)
+    assert device.state == "00:54"
+
+    device = time_date.TimeDateSensor(hass, "date_time")
+    device._update_internal_state(now)
+    assert device.state == "2017-05-17, 20:54"
+
+    device = time_date.TimeDateSensor(hass, "date_time_utc")
+    device._update_internal_state(now)
+    assert device.state == "2017-05-18, 00:54"
+
+    device = time_date.TimeDateSensor(hass, "beat")
+    device._update_internal_state(now)
+    assert device.state == "@079"
+
+    device = time_date.TimeDateSensor(hass, "date_time_iso")
+    device._update_internal_state(now)
+    assert device.state == "2017-05-17T20:54:00"
+
+
+# pylint: disable=no-member
+async def test_timezone_intervals(hass):
+    """Test date sensor behavior in a timezone besides UTC."""
+    new_tz = dt_util.get_time_zone("America/New_York")
+    assert new_tz is not None
+    dt_util.set_default_time_zone(new_tz)
+
+    device = time_date.TimeDateSensor(hass, "date")
+    now = dt_util.utc_from_timestamp(50000)
+    next_time = device.get_next_interval(now)
+    # start of local day in EST was 18000.0
+    # so the second day was 18000 + 86400
+    assert next_time.timestamp() == 104400
+
+    new_tz = dt_util.get_time_zone("America/Edmonton")
+    assert new_tz is not None
+    dt_util.set_default_time_zone(new_tz)
+    now = dt_util.parse_datetime("2017-11-13 19:47:19-07:00")
+    device = time_date.TimeDateSensor(hass, "date")
+    next_time = device.get_next_interval(now)
+    assert next_time.timestamp() == dt_util.as_timestamp("2017-11-14 00:00:00-07:00")
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6356951)